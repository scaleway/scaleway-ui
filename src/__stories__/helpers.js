export const showAlert = (text = 'fired!') => {
  // eslint-disable-next-line no-alert
  alert(text)
}

export const log = (text = 'fired!') => {
  // eslint-disable-next-line no-console
  console.log(text)
}
