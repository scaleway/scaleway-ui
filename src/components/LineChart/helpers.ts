import { Theme } from '@emotion/react'
import { DatumValue, Serie } from '@nivo/line'

const parse = (data?: DatumValue | null) => {
  if (typeof data === 'number') return data
  if (typeof data === 'string') return parseFloat(data)
  if (data instanceof Date) return data.getTime()

  return 0
}

export const getMaxChartValue = (preppedData?: Serie[]): number => {
  if (!preppedData?.length) return 0

  const maximum = Math.max(
    ...preppedData.map(data =>
      data.data.reduce(
        (max, { y }) => Math.max(parse(y), max),
        Number.NEGATIVE_INFINITY,
      ),
    ),
  )

  return Math.ceil(maximum + maximum * 0.1)
}

export const getMinChartValue = (preppedData?: Serie[]): number => {
  if (!preppedData?.length) return 0

  const minimum = Math.min(
    ...preppedData.map(data =>
      data.data.reduce(
        (min, { y }) => Math.min(parse(y), min),
        Number.POSITIVE_INFINITY,
      ),
    ),
  )

  return Math.floor(minimum - minimum * 0.1)
}

export const getMin = (values: number[] = []): number =>
  values.length ? Math.min(...values) : 0

export const getMax = (values: number[] = []): number =>
  values.length ? Math.max(...values) : 0

export const getAverage = (values: number[] = []): number =>
  values.length
    ? Math.round(
        (values.reduce((average, curr) => average + curr, 0) / values.length) *
          100,
      ) / 100
    : 0

export const getCurrent = (values: number[] = []): number =>
  values.length ? values[values.length - 1] : 0

export const getSelected = (
  label: string,
  index: number,
  selected: string[] = [],
): string[] => {
  const labelIndexed = label + index.toString()
  const found = selected.indexOf(labelIndexed)

  if (found > -1) {
    selected.splice(found, 1)
  } else {
    selected.push(labelIndexed)
  }

  return selected
}

export const legendColors = (theme: Theme): string[] =>
  [
    theme.colors.chartGreen,
    theme.colors.chartPurple,
    theme.colors.gray350,
  ] as string[]

export const getLegendColor = (index: number, theme: Theme): string => {
  const colors = legendColors(theme)

  return colors[index] || colors[colors.length - 1]
}
