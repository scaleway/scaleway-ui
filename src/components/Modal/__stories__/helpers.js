export const animations = [
  'fold',
  'zoom',
  'scaleUp',
  'scaleBack',
  'sketch',
  'slideTop',
  'slideBottom',
  'slideLeft',
  'slideRight',
]

export const widths = ['xxsmall', 'xsmall', 'small', 'medium', 'large']

export const placements = [
  'center',
  'top-left',
  'top',
  'top-right',
  'right',
  'bottom-right',
  'bottom',
  'bottom-left',
  'left',
]
