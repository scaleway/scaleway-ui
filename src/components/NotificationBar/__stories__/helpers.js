export default [
  'security',
  'info',
  'success',
  'warning',
  'warning-blue',
  'warning-orange',
  'alert-orange',
  'alert-red',
]
