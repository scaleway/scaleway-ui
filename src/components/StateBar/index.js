import { css } from '@emotion/core'
import { Box } from '@smooth-ui/core-em'
import PropTypes from 'prop-types'
import React from 'react'
import { ProgressBar } from '../ProgressBar'
import { Typography } from '../Typography'

const line = css`
  flex: 1;
  margin-top: 12px;
`

export function StateBar(props) {
  return <Box {...props} />
}

StateBar.State = function State({ label, children, ...props }) {
  return (
    <Typography as="div" variant="bodyA" color="gray950" {...props}>
      <strong>{`${label}${children ? ': ' : ''}`}</strong>
      {children}
    </Typography>
  )
}

StateBar.State.propTypes = {
  label: PropTypes.string,
}

StateBar.Bar = function Bar({ unlimited, value, ...props }) {
  return (
    <ProgressBar
      css={line}
      variant={unlimited ? 'success' : value >= 90 ? 'warning' : 'primary'}
      value={unlimited ? 100 : value}
      {...props}
    />
  )
}

StateBar.Bar.propTypes = {
  unlimited: PropTypes.bool,
  value: PropTypes.number,
}
