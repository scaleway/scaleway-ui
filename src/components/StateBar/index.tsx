import styled from '@emotion/styled'
import type { ReactNode } from 'react'
import { useMemo } from 'react'
import ProgressBar from '../ProgressBar'
import Stack from '../Stack'
import Text from '../Text'

type StateBarStateProps = {
  children?: ReactNode
  label?: string
}

const StyledText = styled(Text)`
  display: flex;
  flex-direction: row;
  gap: ${({ theme }) => theme.space['1']};
`

export const StateBarState = ({
  label = '',
  children,
}: StateBarStateProps): JSX.Element => (
  <StyledText variant="bodyStrong" prominence="strong" as="div" color="neutral">
    <strong>{`${label}${children ? ': ' : ''}`}</strong>
    {children && (
      <Text as="span" variant="body" color="neutral">
        {children}
      </Text>
    )}
  </StyledText>
)

type StateBarBarProps = {
  unlimited?: boolean
  value?: number
  progress?: boolean
}

const StyledProgressBar = styled(ProgressBar)`
  width: 100%;
`

export const StateBarBar = ({
  unlimited = false,
  value = 0,
  progress = false,
}: StateBarBarProps): JSX.Element => {
  const variant = useMemo(() => {
    if (unlimited) return 'success'
    if (value < 90 && value >= 70) return 'warning'
    if (value >= 90) return 'danger'

    return 'primary'
  }, [unlimited, value])

  return (
    <StyledProgressBar
      variant={variant}
      value={unlimited ? 100 : value}
      progress={progress}
    />
  )
}

type StateBarType = ((props: {
  children: ReactNode
  className?: string
}) => JSX.Element) & {
  Bar: (props: StateBarBarProps) => JSX.Element
  State: (props: StateBarStateProps) => JSX.Element
}

const StateBar: StateBarType = ({ children, className }) => (
  <Stack gap={1} className={className}>
    {children}
  </Stack>
)

StateBar.Bar = StateBarBar
StateBar.State = StateBarState

export default StateBar
