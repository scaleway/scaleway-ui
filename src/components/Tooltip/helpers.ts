import { RefObject } from 'react'

export type TooltipPlacement = 'top' | 'right' | 'bottom' | 'left' | 'auto'
export const ARROW_WIDTH = 6 // in px
export const SPACE = 6 // in px
export const TOTAL_USED_SPACE = ARROW_WIDTH + SPACE // in px
export const DEFAULT_POSITIONS = {
  arrowLeft: 50,
  arrowTop: 99,
  arrowTransform: 'translate(-50%, -50)',
  left: 0,
  rotate: 135,
  tooltipInitialPosition: 'translate3d(0, 0, 0))',
  top: 0,
}

export const TOOLTIP_INITIAL_POSITION = {
  auto: `translate3d(0, ${TOTAL_USED_SPACE}px, 0)`,
  bottom: `translate3d(0, -${TOTAL_USED_SPACE}px, 0)`,
  left: `translate3d(${TOTAL_USED_SPACE}px, 0, 0)`,
  right: `translate3d(-${TOTAL_USED_SPACE}px, 0, 0)`,
  top: `translate3d(0, ${TOTAL_USED_SPACE}px, 0)`,
}

type ComputePlacementTypes = {
  childrenStructuredRef: DOMRect
  tooltipStructuredRef: DOMRect
}

/**
 * This function will find the best placement in a window for tooltip based on children position and tooltip size
 */
export const computePlacement = ({
  childrenStructuredRef,
  tooltipStructuredRef,
}: ComputePlacementTypes) => {
  const {
    top: childrenTop,
    left: childrenLeft,
    right: childrenRight,
  } = childrenStructuredRef

  const { width: tooltipWidth, height: tooltipHeight } = tooltipStructuredRef

  if (childrenTop - tooltipHeight - TOTAL_USED_SPACE < 0) {
    return 'bottom'
  }

  if (childrenLeft - tooltipWidth - TOTAL_USED_SPACE < 0) {
    return 'right'
  }

  if (childrenRight + tooltipWidth + TOTAL_USED_SPACE > window.innerWidth) {
    return 'left'
  }

  return 'top'
}

type ComputePositionsTypes = {
  placement: TooltipPlacement
  childrenRef: RefObject<HTMLDivElement>
  tooltipRef: RefObject<HTMLDivElement>
}

/**
 * This function will compute the positions of tooltip and arrow based on children position and tooltip size
 */
export const computePositions = ({
  placement,
  childrenRef,
  tooltipRef,
}: ComputePositionsTypes) => {
  const childrenStructuredRef = (
    childrenRef.current as HTMLDivElement
  ).getBoundingClientRect()
  const tooltipStructuredRef = (
    tooltipRef.current as HTMLDivElement
  ).getBoundingClientRect()

  const placementBasedOnWindowSize =
    placement === 'auto'
      ? computePlacement({
          childrenStructuredRef,
          tooltipStructuredRef,
        })
      : placement

  const {
    top: childrenTop,
    left: childrenLeft,
    right: childrenRight,
    width: childrenWidth,
    height: childrenHeight,
  } = childrenStructuredRef

  const { width: tooltipWidth, height: tooltipHeight } = tooltipStructuredRef

  switch (placementBasedOnWindowSize) {
    case 'bottom':
      return {
        arrowLeft: tooltipWidth / 2,
        arrowTop: -ARROW_WIDTH - 5,
        arrowTransform: '',
        left: childrenLeft + childrenWidth / 2 - tooltipWidth / 2,
        rotate: 180,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.bottom,
        top: childrenTop + childrenHeight + ARROW_WIDTH + SPACE,
      }
    case 'left':
      return {
        arrowLeft: tooltipWidth + ARROW_WIDTH + 5,
        arrowTop: tooltipHeight / 2,
        arrowTransform: 'translate(-50%, -50%)',
        left: childrenLeft - tooltipWidth - ARROW_WIDTH - SPACE * 2,
        rotate: -90,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.left,
        top: childrenTop - tooltipHeight / 2 + childrenHeight / 2,
      }
    case 'right':
      return {
        arrowLeft: -ARROW_WIDTH - 5,
        arrowTop: tooltipHeight / 2,
        arrowTransform: 'translate(50%, -50%)',
        left: childrenRight + ARROW_WIDTH + SPACE * 2,
        rotate: 90,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.right,
        top: childrenTop - tooltipHeight / 2 + childrenHeight / 2,
      }
    default: // top placement is default value
      return {
        arrowLeft: tooltipWidth / 2,
        arrowTop: tooltipHeight - 1,
        arrowTransform: '',
        left: childrenLeft + childrenWidth / 2 - tooltipWidth / 2,
        rotate: 0,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.top,
        top: childrenTop - tooltipHeight - ARROW_WIDTH - SPACE,
      }
  }
}
