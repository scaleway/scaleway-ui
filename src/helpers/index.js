export * from './clipboard'

export const parseIntOr = (str, fallback) => {
  const value = Number.parseInt(str, 10)
  return !Number.isNaN(value) ? value : fallback
}

export const onKeyOnlyNumbers = e => {
  // `e.which` and `e.keyCode` are deprecated (https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent)
  const keyCode = e.key.charCodeAt(0)
  if (keyCode < 48 || keyCode > 57) {
    e.preventDefault()
  }
}
