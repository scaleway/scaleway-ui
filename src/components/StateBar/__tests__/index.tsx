import React from 'react'
import StateBar from '..'
import { shouldMatchEmotionSnapshot } from '../../../helpers/jestHelpers'

describe('StateBar', () => {
  test(`should render correctly with default props`, () =>
    shouldMatchEmotionSnapshot(
      <StateBar>
        <StateBar.State />
        <StateBar.Bar />
      </StateBar>,
    ))

  test(`should render correctly with a children`, () =>
    shouldMatchEmotionSnapshot(
      <StateBar>
        <StateBar.State>Hello</StateBar.State>
        <StateBar.Bar />
      </StateBar>,
    ))

  test(`should render correctly with a value < 90`, () =>
    shouldMatchEmotionSnapshot(
      <StateBar>
        <StateBar.State>Hello</StateBar.State>
        <StateBar.Bar value={89} />
      </StateBar>,
    ))

  test(`should render correctly with a value >= 90`, () =>
    shouldMatchEmotionSnapshot(
      <StateBar>
        <StateBar.State>Hello</StateBar.State>
        <StateBar.Bar value={90} />
      </StateBar>,
    ))

  test(`should render correctly with unlimited value`, () =>
    shouldMatchEmotionSnapshot(
      <StateBar>
        <StateBar.State>Hello</StateBar.State>
        <StateBar.Bar unlimited />
      </StateBar>,
    ))
})
