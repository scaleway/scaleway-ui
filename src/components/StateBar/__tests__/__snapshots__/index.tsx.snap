// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateBar should render correctly with a children 1`] = `
<DocumentFragment>
  .cache-1xgu0ny {
  color: #4a4f62;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
  color: #4a4f62;
  font-size: 16px;
  line-height: 24px;
  color: #151a2d;
}

.cache-1xgu0ny.cache-1xgu0ny {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 500;
}

.cache-1ydlh59 {
  color: #4a4f62;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
  color: #4a4f62;
  font-size: 16px;
  line-height: 24px;
}

.cache-1ydlh59.cache-1ydlh59 {
  margin-left: 8px;
}

.cache-5hex0u-StateBarBar {
  position: relative;
  height: 4px;
  margin-left: 0;
  margin-right: 0;
  border-radius: 2px;
  background-color: #dce1eb;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-top: 12px;
}

.cache-24u8l6 {
  border-radius: 2px;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  background-color: #4f0599;
  -webkit-transition: 0.3s width;
  transition: 0.3s width;
  width: 0%;
}

<div
    class="cache-11djngs e7ah3bn0"
  >
    <div
      class="e1g1zfwd0 cache-1xgu0ny e7ah3bn0"
    >
      <strong>
        : 
      </strong>
      <span
        class="e1g1zfwd0 cache-1ydlh59 e7ah3bn0"
      >
        Hello
      </span>
    </div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      class="et8xa2y2 cache-5hex0u-StateBarBar e7ah3bn0"
      role="progressbar"
    >
      <div
        class="cache-24u8l6 et8xa2y0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`StateBar should render correctly with a value < 90 1`] = `
<DocumentFragment>
  .cache-1xgu0ny {
  color: #4a4f62;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
  color: #4a4f62;
  font-size: 16px;
  line-height: 24px;
  color: #151a2d;
}

.cache-1xgu0ny.cache-1xgu0ny {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 500;
}

.cache-1ydlh59 {
  color: #4a4f62;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
  color: #4a4f62;
  font-size: 16px;
  line-height: 24px;
}

.cache-1ydlh59.cache-1ydlh59 {
  margin-left: 8px;
}

.cache-5hex0u-StateBarBar {
  position: relative;
  height: 4px;
  margin-left: 0;
  margin-right: 0;
  border-radius: 2px;
  background-color: #dce1eb;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-top: 12px;
}

.cache-1fwcdpa {
  border-radius: 2px;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  background-color: #4f0599;
  -webkit-transition: 0.3s width;
  transition: 0.3s width;
  width: 89%;
}

<div
    class="cache-11djngs e7ah3bn0"
  >
    <div
      class="e1g1zfwd0 cache-1xgu0ny e7ah3bn0"
    >
      <strong>
        : 
      </strong>
      <span
        class="e1g1zfwd0 cache-1ydlh59 e7ah3bn0"
      >
        Hello
      </span>
    </div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="89"
      class="et8xa2y2 cache-5hex0u-StateBarBar e7ah3bn0"
      role="progressbar"
    >
      <div
        class="cache-1fwcdpa et8xa2y0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`StateBar should render correctly with a value >= 90 1`] = `
<DocumentFragment>
  .cache-1xgu0ny {
  color: #4a4f62;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
  color: #4a4f62;
  font-size: 16px;
  line-height: 24px;
  color: #151a2d;
}

.cache-1xgu0ny.cache-1xgu0ny {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 500;
}

.cache-1ydlh59 {
  color: #4a4f62;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
  color: #4a4f62;
  font-size: 16px;
  line-height: 24px;
}

.cache-1ydlh59.cache-1ydlh59 {
  margin-left: 8px;
}

.cache-5hex0u-StateBarBar {
  position: relative;
  height: 4px;
  margin-left: 0;
  margin-right: 0;
  border-radius: 2px;
  background-color: #dce1eb;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-top: 12px;
}

.cache-9vex8i {
  border-radius: 2px;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  background-color: #ef5774;
  -webkit-transition: 0.3s width;
  transition: 0.3s width;
  width: 90%;
}

<div
    class="cache-11djngs e7ah3bn0"
  >
    <div
      class="e1g1zfwd0 cache-1xgu0ny e7ah3bn0"
    >
      <strong>
        : 
      </strong>
      <span
        class="e1g1zfwd0 cache-1ydlh59 e7ah3bn0"
      >
        Hello
      </span>
    </div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="90"
      class="et8xa2y2 cache-5hex0u-StateBarBar e7ah3bn0"
      role="progressbar"
    >
      <div
        class="cache-9vex8i et8xa2y0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`StateBar should render correctly with default props 1`] = `
<DocumentFragment>
  .cache-1xgu0ny {
  color: #4a4f62;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
  color: #4a4f62;
  font-size: 16px;
  line-height: 24px;
  color: #151a2d;
}

.cache-1xgu0ny.cache-1xgu0ny {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 500;
}

.cache-5hex0u-StateBarBar {
  position: relative;
  height: 4px;
  margin-left: 0;
  margin-right: 0;
  border-radius: 2px;
  background-color: #dce1eb;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-top: 12px;
}

.cache-24u8l6 {
  border-radius: 2px;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  background-color: #4f0599;
  -webkit-transition: 0.3s width;
  transition: 0.3s width;
  width: 0%;
}

<div
    class="cache-11djngs e7ah3bn0"
  >
    <div
      class="e1g1zfwd0 cache-1xgu0ny e7ah3bn0"
    >
      <strong />
    </div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      class="et8xa2y2 cache-5hex0u-StateBarBar e7ah3bn0"
      role="progressbar"
    >
      <div
        class="cache-24u8l6 et8xa2y0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`StateBar should render correctly with unlimited value 1`] = `
<DocumentFragment>
  .cache-1xgu0ny {
  color: #4a4f62;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
  color: #4a4f62;
  font-size: 16px;
  line-height: 24px;
  color: #151a2d;
}

.cache-1xgu0ny.cache-1xgu0ny {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 500;
}

.cache-1ydlh59 {
  color: #4a4f62;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
  color: #4a4f62;
  font-size: 16px;
  line-height: 24px;
}

.cache-1ydlh59.cache-1ydlh59 {
  margin-left: 8px;
}

.cache-5hex0u-StateBarBar {
  position: relative;
  height: 4px;
  margin-left: 0;
  margin-right: 0;
  border-radius: 2px;
  background-color: #dce1eb;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-top: 12px;
}

.cache-tsbjjr {
  border-radius: 2px;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  background-color: #45d6b5;
  -webkit-transition: 0.3s width;
  transition: 0.3s width;
  width: 100%;
}

<div
    class="cache-11djngs e7ah3bn0"
  >
    <div
      class="e1g1zfwd0 cache-1xgu0ny e7ah3bn0"
    >
      <strong>
        : 
      </strong>
      <span
        class="e1g1zfwd0 cache-1ydlh59 e7ah3bn0"
      >
        Hello
      </span>
    </div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="100"
      class="et8xa2y2 cache-5hex0u-StateBarBar e7ah3bn0"
      role="progressbar"
    >
      <div
        class="cache-tsbjjr et8xa2y0"
      />
    </div>
  </div>
</DocumentFragment>
`;
